.travis.yml

```diff
@@ -0,0 +1,40 @@
+language: c
+dist: trusty
+
+cache:
+  directories:
+    - ~/docker
+
+matrix:
+  include:
+  - env: BASE=ubuntu IMAGE=ubuntu:16.04 CCO=gcc
+
+  - env: BASE=ubuntu IMAGE=ubuntu:16.04 CCO=clang
+
+  - env: BASE=ubuntu IMAGE=ubuntu:16.10 CCO=gcc
+
+  - env: BASE=fedora IMAGE=fedora:25    CCO=gcc
+
+before_install:
+  - cd ..
+  - git clone --depth 1 https://github.com/GNOME/jhbuild.git
+
+before_script:
+  - export NAME=$(echo $IMAGE | sed "s/:/_/g")
+  - ls -lah ~/docker
+
+  # Check if there is a saved base image
+  - if [[ -f ~/docker/$NAME.tar.gz ]]; then cat ~/docker/$NAME.tar.gz | docker import - local/$NAME; fi
```

Comment:
Someone told me once that `<infile some_command` is better than `cat infile | some_command` because it uses fewer resources... but probably doesn't matter too much here. But why not `docker import ~/docker/$NAME.tar.gz local/$NAME` though?